---
title: "Stoichiogenomics"
author: "Jay T. Lennon and Daniel Schwartz"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
  - \usepackage{array}
output: pdf_document
geometry: margin=2.54cm
---

# Set working environment and load packages

```{r setup, message = FALSE, warning = FALSE}
# Clear and set working directory 
rm(list = ls())
library(tidyverse)
library(here)
library(cowplot)
setwd(here())
```

------
# summarize mutations with breseq  
Following commands were run on carbonate

$ module load breseq
  # Curl version 7.54.0 loaded.
  # Sun/Oracle Java SE Development Kit version 11.0.2 loaded.
  # R version 3.6.0 loaded
  # bowtie2 version 2.4.2 loaded.
  # breseq version 0.32.0 loaded.

$ cd ~/GitHub/genetic-dets/breseq

$ gdtools COMPARE -o compare-all.tsv -f TSV -r syn-ancestor.gff genome_diff/*.gd
-------

# Import gd compare table

```{r, message=FALSE}
setwd("~/GitHub/stoichiogenomics/")
#d.compare <- read_tsv(here("breseq/compare-all.tsv"))
d.compare <- read.table(file = "breseq/compare-all.tsv", sep = '\t', header = TRUE)

# select important columns
d <- d.compare %>%
  select(position, type, mutation_category, strain = title)

# convert to wide format
d.wide <- d %>%
  #presence column
  mutate(presence=1)%>%
  pivot_wider(names_from = strain, values_from = presence, values_fill = 0)
  
head(d.wide[,1:8])
```

# Filter relevant mutations

```{r}
# visualize mutations 
d %>%
  ggplot(aes(y = fct_rev(strain), x = position))+
    geom_jitter(aes(fill = type), shape = 21, height = 0.2, width = 5000)+
  theme_bw()
```

There are several positions with mutations in nearly all strains. 
In two of these loci (positions ~500K and ~ 2M) there are multiple mutations per strain. 
The 500K locus and the right most locus have a similar pattern: 
they disappear in all strains sequenced from day 21, irrespective of chemostat and treatment.

```{r}
#visualize 500k
p1 <- 
d %>%
  filter(position>5.3e5)%>%
  filter(position<5.4e5)%>%
  ggplot(aes(y=fct_rev(strain), x=position))+
    geom_jitter(aes(fill=type), shape=21, height = 0.3,width = 50)+
  theme_bw()+
  theme(legend.position = "bottom")

#visualize 2M
p2 <- 
d %>%
  filter(position>2.01e6)%>%
  filter(position<2.03e6)%>%
  ggplot(aes(y=fct_rev(strain), x=position))+
    geom_jitter(aes(fill=type), shape=21, height = 0.3,width = 50)+
  theme_bw()+
  theme(legend.position = "bottom")

plot_grid(p1,p2)
```

# Summarize mutations by gene for filtering

```{r}
d.gene <- d.compare%>%
  select(gene_name, strain=title)%>%
  group_by(gene_name, strain)%>%
  summarise(n=n(), .groups="drop")%>%
  arrange(desc(n))

# visualize
d.gene %>%
  # mutate(gene_product = str_trunc(gene_product,30)%>%fct_infreq()%>%fct_rev())%>%
  mutate(gene_name = fct_infreq(gene_name)%>%fct_rev())%>%
  ggplot(aes(x = strain, y = gene_name))+
  geom_tile(aes(fill = n))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

# Visulaize filtering at different levels of mutation prevalence within strain collection

```{r}
# list of plots
l.plot <- list()

# try several cutoffs
n.max <- c(1, 0.8, 0.6, 0.4, 0.2)

for (i in n.max){
  #filter based on cutoff
  filt.out <- sum.gene %>%
  filter(avg.mut.n>i) %>%
  pull(gene_name)%>%
  unique()

  l.plot[[as.character(i)]] <- d.compare %>%
    filter(! gene_name %in% filt.out) %>%
  select(position, type, mutation_category, strain=title) %>%
    #plot
  ggplot(aes(y=fct_rev(strain), x=position))+
    geom_jitter(aes(fill = type), shape = 21, height = 0.2, width = 5000)+
  theme_bw()+
  expand_limits(x = c(0, max(d.compare$position)))
}

plot_grid(plotlist = l.plot, nrow = 5, labels = n.max, label_colour = "red", label_x = 0.1)
```

# Export data for contingency analysis

```{r}
d.write <- left_join(d.compare, sum.gene) %>%
  select(strain_ID = title, gene=locus_tag, gene_name, position, snp_type, 
         type, prevalence = avg.mut.n) %>%
  # add metadata
  mutate(Time = str_extract(strain_ID, "^..") %>% parse_integer()) %>%
  mutate(cID=str_extract(strain_ID, "N.|P.")) %>%
  mutate(clone=str_extract(strain_ID, "S."))

write_csv(d.write, here("data/mutPositions_wPrevalence.csv"))
```

(# what is purpose of dashed line on following line??) 
--------------- 
Summarize mutations by position for filtering

```{r}
d.pos <- d.compare %>%
  select(position, strain=title) %>%
  group_by(position, strain) %>%
  summarise(n=n(), .groups="drop") %>%
  arrange(desc(n))

#visualize
d.pos %>%
  # mutate(gene_product = str_trunc(gene_product,30)%>%fct_infreq()%>%fct_rev())%>%
  mutate(position =as_factor(position) %>% fct_infreq() %>% fct_rev()) %>%
  ggplot(aes(x=strain, y=position))+
  geom_tile()+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5))
```

# Visulaize filtering at different levels of mutation prevalence within the strain collection

```{r}
#list of plots
l.plot <- list()

# try several cutoffs
n.max <- c(1,0.8,0.6,0.4, 0.2)

for (i in n.max){
  #filter based on cutoff
  filt.out <- sum.pos %>%
  filter(avg.mut.n>i) %>%
  pull(position)%>%
  unique()

  l.plot[[as.character(i)]] <- d.compare %>%
    filter(! position %in% filt.out) %>%
  select(position, type, mutation_category, strain=title) %>%
    #plot
  ggplot(aes(y=fct_rev(strain), x=position))+
    geom_jitter(aes(fill=type), shape=21, height = 0.2,width = 5000)+
  theme_bw()+
  expand_limits(x=c(0, max(d.compare$position)))
}


plot_grid(plotlist = l.plot, nrow = 5, labels = n.max, label_colour = "red", label_x = 0.1)
```


